README.md

```diff
@@ -14,6 +14,30 @@ meson setup builddir
 ninja -C builddir test
 ```
 
+Meson currently is configured with support to build all of it's dependencies
+from source if they are not otherwise detected, by fetching at configure time.
+If this is not desirable, calling:
+
+```sh
+meson setup builddir --wrap-mode=nofallback
+```
+
+Will disable this behavior. This will require that all dependencies have been
+installed and are discoverable at configure time.
+
+Test execution can be controlled by setting the `test` option to Meson:
+```sh
+meson setup builddir -Dtests=disabled
+```
+or to reconfigure
+```sh
+meson configure builddir -Dtests=enabled
+```
+This is a Meson [feature](https://mesonbuild.com/Build-options.html#features)
+option, which can have a state of `disabled`, `enabled`, or `auto`, as described
+in the linked documentation. The default is `auto`, which is usually desirable
+for end users.
```

Comment:
Are you running into the issue of:
```sh
meson setup builddir
git pull
meson configure builddir -Dtest=...
```
?

That's an annoying one that we've been going back and forth on for a while, but you can fix that by replacing `configure` with `setup --reconfigure`:
```sh
meson setup builddir --reconfigure

Reply:
That only happens if you setup the builddir without an option, do a pull, and don't reconfigure first (which calling ninja will do automatically), and is a side effect of Meson rejecting options it doesn't know about. Which was a trade-off between "passing invalid arguments and being confused why they didn't work" and this situation

Reply:
I don't recall exactly, but the Meson docs to state that certain kinds of decisions are baked into the build directory upon configuration and are not revised. It recommends reconfiguring completely when you change certain kinds of things. Also the subprojects directory right now goes in the source directory, but I'm not sure if that's optional or what. CMake FetchContent puts downloaded dependencies in the build directory.

Reply:
Regardless, we should keep it pretty simple. I thought recommending a new build dir would work and would be simple, but if I'll trust your judgement on that detail.

Reply:
Only options that would invalidate the entire build directory are supposed to be "baked in" things like "Which C++ compiler do you want, want backend do you want?" If we're running into other cases that sounds like a bug. I'm tempted to write code that would allow `meson configure -Dfoo=bar` to update the stored options now...

The only thing that's put in the source directory is the fetched sources, this is an optimization so that if you run multiple builds from the same source repo you don't have to re-fetch the sources. All of the built artifacts go into $builddir/subprojects/*. Now that you mention it, we might need to handle the source dir being immutable... Although, on the other hand `meson dist` can pack those sources into the dist tarball, so maybe it's not that important? Not sure

Reply:
The answer from me is: I'm not sure about the wiping? I use Meson so often it doesn't seem like an issue, but maybe it's unexpected for people coming from CMake